src/page_managers/match_manager.py

```diff
@@ -62,3 +62,131 @@ def generate_input_section(self) -> list[list, list]:
             [blue_1, blue_2, blue_3]
         ]
+    
+    def generate_graphs(self, team_numbers: list, display_points: bool = False) -> None:
+        """Generates the graphs for the `Match` page.
+
+        :param team_numbers: The teams to generate the graphs for.
+        :return:
+        """
+        
+        teams_data = [scouting_data_for_team(team) for team in team_numbers]
+
+        
+        auto_graphs_tab, teleop_graphs_tab, endgame_graphs_tab = st.tabs(
+            ["🤖 Autonomous", "🎮 Teleop", "🧗 Endgame"]
+        )
+
+        # Autonomous graphs
+        with auto_graphs_tab:
+            st.write("#### Autonomous")
+
+            # Graph for auto cycles over time
+            auto_cycles_over_time_per_team = [self.calculated_stats.points_contributed_by_match(team, Queries.AUTO_GRID) if display_points else self.calculated_stats.cycles_by_match(team, Queries.AUTO_GRID) for team in team_numbers]
+
+            st.plotly_chart(
+                multi_line_graph(
+                    x=teams_data[0][Queries.MATCH_KEY],
+                    y=auto_cycles_over_time_per_team,
+                    x_axis_label="Match Key",
+                    y_axis_label=team_numbers,
+                    y_axis_title= "Auto Point Total" if display_points else "# of Auto Cycles"
+                ),
+                use_container_width=True
+            )
+
+            auto_cycles_cone_vs_cube_col, auto_cycles_by_level_col = st.columns(2)
+
+            with auto_cycles_cone_vs_cube_col:
+                auto_cycles_cone_vs_cube = [[self.calculated_stats.cycles_by_game_piece_per_match(team, Queries.AUTO_CONES).mean() for team in team_numbers], [self.calculated_stats.cycles_by_game_piece_per_match(team, Queries.AUTO_CUBES).mean() for team in team_numbers]]
+                st.plotly_chart(
+                    stacked_bar_graph(
+                        x=team_numbers,
+                        y=auto_cycles_cone_vs_cube,
+                        x_axis_label="Teams",
+                        y_axis_label=[
+                            "Cones","Cubes"
+                        ],
+                        y_axis_title="Auto Cones vs Cubes",
+                    ),use_container_width=True
+                )
+
+            with auto_cycles_by_level_col:
+                auto_cycles_by_level = [[self.calculated_stats.cycles_by_height_per_match(team, Queries.AUTO_GRID, Queries.HIGH).mean() for team in team_numbers], [self.calculated_stats.cycles_by_height_per_match(team, Queries.AUTO_GRID, Queries.MID).mean() for team in team_numbers], [self.calculated_stats.cycles_by_height_per_match(team, Queries.AUTO_GRID, Queries.LOW).mean() for team in team_numbers]]
+                st.plotly_chart(
+                    stacked_bar_graph(
+                        x=team_numbers,
+                        y=auto_cycles_by_level,
+                        x_axis_label="Teams",
+                        y_axis_label=[
+                            "High","Mid", "Low"
```

Comment:
small nitpick but keep the whitespace consistent between commas if possible
```suggestion
                            "High", "Mid", "Low"
```